templates/adding_a_new_model/open_model_proposals/ADD_BIG_BIRD.md

```diff
@@ -0,0 +1,1142 @@
+
+Not quite ready yet :-( This section will be added soon!
+
+Step-by-step recipe to add a model to 🤗 Transformers
+----------------------------------------------------
+
+Everyone has different preferences of how to port a model so it can be
+very helpful for you to take a look at summaries of how other
+contributors ported models to Hugging Face. Here is a list of community
+blog posts on how to port a model:
+
+1.  [Porting GPT2
+    Model](https://medium.com/huggingface/from-tensorflow-to-pytorch-265f40ef2a28)
+    by [Thomas](https://huggingface.co/thomwolf)
+2.  [Porting WMT19 MT Model](https://huggingface.co/blog/porting-fsmt)
+    by [Stas](https://huggingface.co/stas)
+
+From experience, we can tell you that the most important things to keep
+in mind when adding a model are:
+
+-   Don\'t reinvent the wheel! Most parts of the code you will add for
+    the new 🤗 Transformers model already exist somewhere in 🤗
+    Transformers. Take some time to find similar, already existing
+    models and tokenizers you can copy from.
+    [grep](https://www.gnu.org/software/grep/) and
+    [rg](https://github.com/BurntSushi/ripgrep) are your friends. Note
+    that it might very well happen that your model\'s tokenizer is based
+    on one model implementation, and your model\'s modeling code on
+    another one. *E.g.* FSMT\'s modeling code is based on BART, while
+    FSMT\'s tokenizer code is based on XLM.
+-   It\'s more of an engineering challenge than a scientific challenge.
+    You should spend more time on creating an efficient debugging
+    environment than trying to understand all theoretical aspects of the
+    model in the paper.
+-   Ask for help, when you\'re stuck! Models are the core component of 🤗
+    Transformers so that we at Hugging Face are more than happy to help
+    you at every step to add your model. Don\'t hesitate to ask if you
+    notice you are not making progress.
+
+In the following, we try to give you a general recipe that we found most
+useful when porting a model to 🤗 Transformers.
+
+The following list is a summary of everything that has to be done to add
+a model and can be used by you as a To-Do List:
+
+1.  [ ] (Optional) Understood theoretical aspects
+
+2.  [ ] Prepared transformers dev environment
+
+3.  [ ] Set up debugging environment of the original repository
+
+4.  [ ] Created script that successfully runs forward pass using
+    original repository and checkpoint
+
+5.  [ ] Successfully added the model skeleton to Transformers
+
+6.  [ ] Successfully converted original checkpoint to Transformers
+    checkpoint
+
+7.  [ ] Successfully ran forward pass in Transformers that gives
+    identical output to original checkpoint
+
+8.  [ ] Finished model tests in Transformers
+
+9.  [ ] Successfully added Tokenizer in Transformers
+
+10. [ ] Run end-to-end integration tests
+
+11. [ ] Finished docs
+
+12. [ ] Uploaded model weights to the hub
+
+13. [ ] Submitted the pull request
```

Comment:
I would argue opening a PR early on would be the best, as it's way easier to help users when they've opened a PR